lib/customer-deposit-wallet/test/unit/Cardano/Wallet/Deposit/HTTP/JSON/JSONSpec.hs

```diff
@@ -26,58 +32,117 @@ import Cardano.Wallet.Deposit.Read
 import Data.Aeson
     ( FromJSON (..)
     , ToJSON (..)
+    , Value
     , decode
     , encode
     )
+import Data.Aeson.Encode.Pretty
+    ( encodePretty
+    )
+import Data.OpenApi
+    ( Definitions
+    , Schema
+    , validateJSON
+    )
 import Test.Hspec
-    ( Spec
+    ( Expectation
+    , Spec
     , describe
     , it
+    , shouldBe
     )
 import Test.QuickCheck
     ( Arbitrary (..)
+    , Gen
     , Property
+    , Testable
     , arbitrarySizedBoundedIntegral
     , chooseInt
+    , counterexample
+    , forAll
     , property
     , shrinkIntegral
     , vectorOf
     , (===)
     )
 
 import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy.Char8 as BL
+import qualified Data.List as L
 
 spec :: Spec
-spec =
+spec = do
     describe "JSON serialization & deserialization" $ do
         it "ApiT Address" $ property $
             prop_jsonRoundtrip @(ApiT Address)
         it "ApiT Customer" $ property $
             prop_jsonRoundtrip @(ApiT Customer)
         it "ApiT CustomerList" $ property $
             prop_jsonRoundtrip @(ApiT CustomerList)
+    describe "schema checks" $ do
+        it "ApiT Address"
+            $ forAll genApiAddress
+            $ counterExampleJSON "validate"
+            $ validateInstance depositDefinitions addressSchema
+        it "ApiT Customer"
+            $ forAll genApiCustomer
+            $ counterExampleJSON "validate"
+            $ validateInstance depositDefinitions customerSchema
+        it "ApiT CustomerList"
+            $ forAll genApiCustomerList
+            $ counterExampleJSON "validate"
+            $ validateInstance depositDefinitions customerListSchema
+
+validate :: Definitions Schema -> Schema -> Value -> Expectation
+validate defs sch x = validateJSON defs sch x `shouldBe` []
+
+validateInstance :: ToJSON a => Definitions Schema -> Schema -> a -> Expectation
+validateInstance defs sch = validate defs sch . toJSON
+
+counterExampleJSON
+    :: (Testable prop, ToJSON a)
+    => String
+    -> (a -> prop)
+    -> a
+    -> Property
+counterExampleJSON t f x =
+    counterexample
+        ("Failed to " <> t <> ":\n" <> BL.unpack (encodePretty $ toJSON x))
+        $ f x
 
 prop_jsonRoundtrip :: (Eq a, Show a, FromJSON a, ToJSON a) => a -> Property
 prop_jsonRoundtrip val =
     decode (encode val) === Just val
 
+genApiAddress :: Gen (ApiT Address)
+genApiAddress = do
+    --enterprise address type with key hash credential is 01100000, network (mainnet) is 1
+    --meaning first byte is 01100001 ie. 96+1=97
+    let firstByte = 97
+    keyhashCred <- BS.pack <$> vectorOf 28 arbitrary
+    pure $ ApiT $ fromRawAddress $ BS.append (BS.singleton firstByte) keyhashCred
```

Comment:
```suggestion
genAddress :: Gen Address
genAddress = do
    --enterprise address type with key hash credential is 01100000, network (mainnet) is 1
    --meaning first byte is 01100001 ie. 96+1=97
    let firstByte = 97
    keyhashCred <- BS.pack <$> vectorOf 28 arbitrary
    pure $ fromRawAddress $ BS.append (BS.singleton firstByte) keyhashCred

genApiTAddress :: Gen (ApiT Address)
genApiTAddress = ApiT <$> genAddress
```

🤷 This seems conceptually clearer to me — in essence, we generate a plain `Address`, we just package it as a `ApiT` for the purpose of controlling the `ToJSON` and `FromJSON` instances.